walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,41 @@
 
 package com.google.sps;
 
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventsList = new ArrayList<>(events);
+    Collection<TimeRange> timeOfMeetings = new ArrayList<>();
+    int potentialMeetingTime = TimeRange.START_OF_DAY;
+    Collections.sort(eventsList, (Event e1, Event e2) ->
+        TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
+    // Schedules potential meetings inbetween the end time of one event to the start time of another.
+    for(Event event: eventsList) {
+      // Meeting times should only be checked if the attendees to the event are relevant to the meeting.
+      if(request.getAttendees().containsAll(event.getAttendees())) {
+        // The end time of one meeting should not be after the start of another + the chunk of time
+        // must properly fit the duration the meeting requires
+        if(potentialMeetingTime < event.getWhen().start() && 
+            event.getWhen().start() - potentialMeetingTime >= request.getDuration()) {
+              timeOfMeetings.add(TimeRange.fromStartEnd(potentialMeetingTime, event.getWhen().start(), false));
+            }
+        // The potential meeting time should only be changed if the end time is greater than the events previous,  
+        // Acts as a safeguard for overlapping events.
+        if(event.getWhen().end() > potentialMeetingTime) {
+            potentialMeetingTime = event.getWhen().end();
+        }
+      }
+    }
+    // There is a potential time slot from the end of the last event to the end of the day that has
+    // to be checekd.
+    if(TimeRange.END_OF_DAY - potentialMeetingTime >= request.getDuration()) {
+      timeOfMeetings.add(TimeRange.fromStartEnd(potentialMeetingTime, TimeRange.END_OF_DAY, true));
+    }
```

Comment:
This particular approach is really hard to read, particularly because you are not making use of the TImeRange API. Consider another approach:
1. Get a list of TimeRanges of all the in between times of the events
2. Look through all these TimeRanges and find the one whos duration is >= the desired duration. 
3. Done